### DISCUSSION TITLE
Node Graceful shutdown questions

### DISCUSSION THREAD

**Discussion by ziluChou** [rabbitmq/rabbitmq-server#10447] [Questions] [answered]
Hi, all.

In a cluster with three nodes, queues are all quorum queues. If a node goes offline(Graceful shutdown), what actions will the node and cluster take? How long will the impact on the cluster's receiving of messages? Am looking for a documentation link for the same.

Thanks in advance

**Comment by michaelklishin** [comment, marked as answer]
* [Clustering](https://www.rabbitmq.com/clustering.html)
* [Quorum queues](https://rabbitmq.com/quorum-queues.html#leader-election)
* [Upgrading](https://rabbitmq.com/upgrade.html), [Marking nodes for maintenance](https://rabbitmq.com/upgrade.html#maintenance-mode)
* [Rebalancing replicas](https://rabbitmq.com/quorum-queues.html#replica-rebalancing)
* An [overview of reliability-related topics](https://rabbitmq.com/reliability.html)
* [Management plugin in clusters](https://rabbitmq.com/management.html#clustering)

**Reply by ziluChou** [reply]
Thanks, I have another question, if the follower node goes offline, will the quorum queue re-elect for a new Leader?

**Reply by michaelklishin** [reply]
[Leader Election and Failure Handling of Quorum Queues](https://rabbitmq.com/quorum-queues.html#leader-election)

**Reply by michaelklishin** [reply]
[Leader election and Failure Handling of Streams](https://rabbitmq.com/streams.html#leader-election)

**Reply by ziluChou** [reply]
Thanks, When a RabbitMQ node hosting a quorum queue's leader fails or is stopped another node hosting one of that quorum queue's follower will be elected leader and resume operations.
So, only  quorum queue's leader node shutdown will result in relection.
